### ISSUE TITLE
timeout: :infinity fails.

### ISSUE THREAD

**Issue by ityonemo** [appcues/mojito#35]
just in case we want to use mojito to download a really big file.

**Comment by christhekeele** [comment]
Just noticed this myself.

I'm messing around with an HTTP Proxy via Mojito and wrapping calls in `Task`s with configurable timeouts, wanted to perform calls with `timeout: :infinity` at the Mojito layer as not to stomp on the caller's preference.

Instead I get an ArithmeticError, which makes sense but is undesirable:

```
** (ArithmeticError) bad argument in arithmetic expression
    :erlang.-(:infinity, 0)
    (mojito) lib/mojito/pool/single.ex:115: Mojito.Pool.Single.request/2
    (mojito) lib/mojito/pool.ex:51: Mojito.Pool.do_request/3
```

**Comment by gamache** [comment]
Thanks for this report! `:infinity` support would be a good feature and I can add it soon.

**Comment by gamache** [comment]
Fixed in 0.6.2.
